co2calculator/mobility/calculate_mobility.py

```diff
@@ -145,55 +152,66 @@ def calc_co2_ferry(
     :param options: Options for the ferry trip
     :type distance: Kilometer
     :type seating: str
-    :return: Total emissions of sea travel in co2 equivalents
+    :return: Total emissions of sea travel in co2 equivalents, Co2e factor and the Parameters
     :rtype: Kilogram
     """
 
     if options is None:
         options = {}
-    params = FerryEmissionParameters(**options)
+    params = FerryEmissionParameters.parse_obj(options)
     # Get the co2 factor
-    co2e = emission_factors.get(params.dict())
-    return distance * co2e
+    co2e_factor = emission_factors.get(params.dict())
+    # Calculate emissions
+    co2e = distance * co2e_factor
+    return co2e, co2e_factor, params
 
 
 def calc_co2_bicycle(distance: Kilometer) -> Kilogram:
     """Calculate co2 emissions for commuting by bicycle
 
     :param distance: distance in km
+    :return: Total emissions of bicycle in co2 equivalents, Co2e factor and the parameters
+    :rtype: Kilogram
     """
-    co2e = emission_factors.get(
+    co2e_factor = emission_factors.get(
         {
             "category": EmissionCategory.TRANSPORT,
             "subcategory": TransportationMode.BICYCLE,
         }
     )
-    return co2e * distance
+    co2e = distance * co2e_factor
+    return co2e, co2e_factor
 
 
 def calc_co2_pedelec(distance: Kilometer) -> Kilogram:
     """Calculate co2 emissions for commuting by pedelec
 
     :param distance: distance in km
+    :return: Total emissions of pedelec in co2 equivalents, Co2e factor and the parameters
+    :rtype: Kilogram
     """
-    co2e = emission_factors.get(
+    co2e_factor = emission_factors.get(
         {
             "category": EmissionCategory.TRANSPORT,
             "subcategory": TransportationMode.PEDELEC,
         }
     )
-    return co2e * distance
+    co2e = distance * co2e_factor
+    return co2e, co2e_factor
 
 
 def calc_co2_tram(distance) -> Kilogram:
     """Calculate co2 emissions for commuting by pedelec
 
     :param distance: distance in km
+    :return: Total emissions of tram in co2 equivalents, Co2e factor and the parameters
+    :rtype: Kilogram
     """
-    co2e = emission_factors.get(
+    co2e_factor = emission_factors.get(
         {
             "category": EmissionCategory.TRANSPORT,
             "subcategory": TransportationMode.TRAM,
         }
     )
-    return co2e * distance
+    co2e = distance * co2e_factor
+    return co2e, co2e_factor
```

Comment:
return None here as a third object so that the length of the returned tuple is always 3